src/linear_combination.jl

```diff
@@ -0,0 +1,59 @@
+"""
+    LinearCombination <: Transform
+
+Calculate the linear combination using the column weights passed in.
+"""
+struct LinearCombination <: Transform
+    col_weights::Vector{Real}
```

Comment:
This is perhaps a nitpick but I believe the proper term for this in a [linear combination context](https://en.wikipedia.org/wiki/Linear_combination) is "coefficients".

I believe "weights" come with additional meaning in ML/statistics of being proportional quantities (probabilities, frequencies, etc.) and usually sum to 1